### ISSUE TITLE
Edit census.ipynb example: Use dask.dataframe.read_hdf vice pd.read_hdf

### ISSUE THREAD

**Issue by linwoodc3** [bokeh/datashader#206]
I referenced this issue in the dask issue tracker, created a pull request for documentation; and am wondering if we should do the same to update the datashader example.  

[Cross reference issue #664](https://github.com/dask/dask/pull/664#issuecomment-234730449) for the write up showing how this will help folks trying to learn `datashader` and `dask`.  Moreover, it makes running the census example easier on computers with little RAM and enables out of core operations since the `.castra` file is no longer available ([discussed in detail with screen shots here](https://github.com/dask/dask/pull/664#issuecomment-234730449))

I apologize if this is the wrong forum; trying to contribute to something I found useful.  #

**Comment by jbednar** [comment]
Thanks for the suggestion, and sorry that it took me a while to get around to looking at it! I think this is definitely a datashader issue rather than one for the dask site.  The castra file is probably still available in the same location (I don't *think* we deleted it), but we stopped having people download it by default because the file only works on Python2.  

Unfortunately, the various file formats and loading options vary a *lot* in performance.  On my 16GB MacBook Pro, I get the following speeds for the three different suggested options:

**castra using dask:** 10 seconds to load, 5 seconds to aggregate
**hdf5 using pandas:** 67 seconds to load, 16 seconds to aggregate
**hdf5 using dask:** 114 seconds to load, 31 seconds to aggregate

Castra+dask is in a class all its own, so it's a big shame that it won't work cross platform.  But as you can see, I get a pretty substantial slowdown from using dask instead of pandas with HDF5, at least on this machine and with the code listed below.  So I'd be reluctant to make that the default way the data is loaded -- people will then think that's the performance of datashader, when really it's about the data itself and how it is stored and accessed.

@jcrist, maybe there is some flag I could pass to dask so that the hdf5/dask version is faster somehow?  I think I heard that the hdf5 conda package was not compiled with multithreading enabled, which presumably limits its performance.  Any other suggestions?  See also issue #129.

```python
%%time
try:
    import dask.dataframe as dd
    df = dd.from_castra('data/census.castra')
    df = df.cache(cache=dict)
    print("Read using dask from castra")
except:
    try:
        df = dd.read_hdf('./data/census.h5', '/census',chunksize=10000000)
        df.categorize(columns=['race']).compute()
        print("Read using dask from HDF5")
    except:
        df = pd.read_hdf('data/census.h5', 'census')
        df.race = df.race.astype('category')
        print("Read using pandas from HDF5")
```

**Comment by linwoodc3** [comment]
Makes sense to me; I was almost maxed out on RAM for the pandas example.  But, performance explanation makes sense.

By the way, your YouTube videos where you say downsampling can lead to incorrect assumptions on the data is correct.  I plotted a full data set using datashader and saw some biases (holes) I never would have seen if I were only downsampling.  

This is a fantastic tool.
